### PR TITLE
Store Monitoring v2 Changes

### PR DESCRIPTION
**Pull request by Yibaebi**

### CODE REVIEW

File: tests/integration/components/app-monitoring/version-table/countries-test.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
+import { hbs } from 'ember-cli-htmlbars';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { setupRenderingTest } from 'ember-qunit';
+import { module, test } from 'qunit';
+import { COUNTRY_NAMES_MAP } from 'irene/utils/constants';
+import {
```
Comment (future-pirate-king) [inline tests/integration/components/app-monitoring/version-table/countries-test.js:6]: add a new line above

File: tests/integration/components/app-monitoring/version-table/index-test.js [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+            am_app_store_instance: amAppStoreInstances[idx].id,
+          })
+        );
+
+        const normalizedAmAppVersion = this.store.normalize(
+          'am-app-version',
+          amAppVersion.toJSON()
+        );
+
+        return this.store.push(normalizedAmAppVersion);
```
Comment (future-pirate-king) [inline tests/integration/components/app-monitoring/version-table/index-test.js:58]: here also component only requires `amApp` so this should be resolved internally

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/adapters/am-app.ts [hunk 3]
```diff
@@ -10,6 +14,24 @@ export default class AmAppAdapter extends commondrf {
...
+    store: Store,
+    type: ModelSchema<K>,
+    snapshot: Snapshot<K>
+  ) {
+    const data = {};
+    // @ts-expect-error to be fixed
+    const serializer = store.serializerFor(type.modelName);
+
+    // @ts-expect-error to be fixed
+    serializer.serializeIntoHash(data, type, snapshot);
```
[the remark this answers is not in the document]
Comment (Yibaebi) [reply app/adapters/am-app.ts:28]: Yes. It is required for the update call in line 33.

File: app/components/app-monitoring/version-table/actions/index.ts [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+      this.relatedVersionSubmissionRecord = await this.amAppVersion.get(
+        'uploadSubmission'
+      );
+
+      addObserver(
+        this.relatedVersionSubmissionRecord,
+        'status',
+        this,
+        this.reloadAmAppVersion
+      );
```
[the remark this answers is not in the document]
Comment (Yibaebi) [reply app/components/app-monitoring/version-table/actions/index.ts:105]: Resolved internally.

File: app/components/app-monitoring/version-table/countries/index.hbs [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{{#if this.fetchAmAppVersionRecords.isRunning}}
+  <AkLoader @size={{16}} data-test-details-table-countriesCodesLoader />
+{{else}}
+  <AkTypography data-test-details-table-countriesCodes>
+    {{#if this.showEmptyCountryList}}
+      -
+    {{else}}
+      {{#each this.this.countryCodes as |countryCode idx|}}
```
Comment (future-pirate-king) [inline app/components/app-monitoring/version-table/countries/index.hbs:8]: typo `this.countryCodes`

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/components/app-monitoring/version-table/countries/index.hbs (hunk 5 above)
Comment (Yibaebi) [reply app/components/app-monitoring/version-table/countries/index.hbs:8]: Fixed. Thanks.

File: app/models/submission.ts [hunk 6]
```diff
@@ -24,6 +26,8 @@ export const submissionStatusProgressPercentage = {
 };
 
 export default class SubmissionModel extends ModelBaseMixin {
+  @service declare intl: IntlService;
```
Comment (future-pirate-king) [inline app/models/submission.ts:29]: this is not being used

File: translations/en.json [hunk 7]
```diff
@@ -753,6 +802,7 @@
   "nistsp80053": "NIST SP 800-53",
   "nistsp800171": "NIST SP 800-171",
   "no": "No",
+  "noActionRequired": "<em>No Action Required</em>",
```
Comment (future-pirate-king) [inline translations/en.json:805]: `em` tag can be wrapped at the usage point

File: translations/ja.json [hunk 8]
```diff
@@ -753,6 +802,7 @@
   "nistsp80053": "NIST SP 800-53",
   "nistsp800171": "NIST SP 800-171",
   "no": "いいえ",
+  "noActionRequired": "<em>No Action Required</em>",
```
Comment (future-pirate-king) [inline translations/ja.json:805]: `em` tag can be wrapped at the usage point

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/models/submission.ts (hunk 6 above)
Comment (Yibaebi) [reply app/models/submission.ts:29]: Removed

File: translations/en.json (hunk 7 above)
Comment (Yibaebi) [reply translations/en.json]: Done

**Review by future-pirate-king — APPROVED** [review]

**Pull request merged**